satellite/metainfo/metainfo.go

```diff
@@ -475,18 +478,38 @@ func (endpoint *Endpoint) filterValidPieces(ctx context.Context, pointer *pb.Poi
 	defer mon.Task()(&ctx)(&err)
 
 	if pointer.Type == pb.Pointer_REMOTE {
-		var remotePieces []*pb.RemotePiece
 		remote := pointer.Remote
+
+		nodeIDList := storj.NodeIDList{}
```

Comment:
The section that starts here and ends on line 494 is very nicely self-contained (the only inputs are `endpoint`, `ctx`, and `remote.RemotePieces`, and the only output is `peerIDMap` (and, ok, `err`). This should be broken out to a helper method with a signature something like

```go
func (endpoint *Endpoint) mapNodesFor(ctx context.Context, pieces []*pb.RemotePiece) (map[storj.NodeID]*identity.PeerIdentity, error) {
```

Reply:
Done